markup/goldmark/codeblocks/integration_test.go

```diff
@@ -265,6 +265,39 @@ Position: {{ .Position | safeHTML }}
 	b.AssertFileContent("public/p1/index.html", filepath.FromSlash("Position: \"/content/p1.md:7:1\""))
 }
 
+// Issue 10118
+func TestAttributes(t *testing.T) {
+	t.Parallel()
+
+	files := `
+-- config.toml --
+-- content/p1.md --
+---
+title: "p1"
+---
+
+## Issue 10118
+
+§§§ {foo="bar"}
+Hello, World!
+§§§
+
+-- layouts/_default/single.html --
+{{ .Content }}
+-- layouts/_default/_markup/render-codeblock.html --
+Attributes: {{ .Attributes }}
```

Comment:
Could you

* Add a check for the `Type` and make sure it's empty.
* Get the commit message title in line with the guidelines ("get" => "Get")

Reply:
@bep 
I'd fixed the commit message, sorry.

I have a question.
I tried to add to check `Type`, but the `Type` returns Attributes.
<img width="1027" alt=" 2022-07-24 8 59 24" src="https://user-images.githubusercontent.com/14119304/180627042-7f87c120-7685-4074-b00e-74ad56d9940b.png">

Because the parser returns a string of after the code fence if a language identifier is empty.
https://github.com/yuin/goldmark/blob/master/ast/block.go#L273-L276

Would it be better to resolve the problem on this pull request or another Issue?

Reply:
I'm not sure I follow, it looks like a string to me:

https://github.com/gohugoio/hugo/blob/9c24b86e4b634d7db761a3e57c1b64fead880aff/markup/goldmark/codeblocks/render.go#L158

Reply:
Sorry. The order of my comment was not good.

After adding `.Type`, a result of `TestAttributes()` as follows:
```
--- FAIL: TestAttributes (0.04s)
    /Users/chick-p/Repos/hugo/markup/goldmark/codeblocks/integration_test.go:298: 
        error:
          no substring match found
        comment:
          <h2 id="issue-10118">Issue 10118</h2>
          Attributes: map[foo:bar]|Type: |
        container:
          "<h2 id=\"issue-10118\">Issue 10118</h2>\nAttributes: map[foo:bar]|Type: {foo=&#34;bar&#34;}|"
        want:
          "Attributes: map[foo:bar]|Type: |"
              /Users/chick-p/Repos/hugo/hugolib/integrationtest_builder.go:144
            s.Assert(content, qt.Contains, match, qt.Commentf(m))
              /Users/chick-p/Repos/hugo/markup/goldmark/codeblocks/integration_test.go:298
            b.AssertFileContent("public/p1/index.html", "<h2 id=\"issue-10118\">Issue 10118</h2>\nAttributes: map[foo:bar]|Type: |")
FAIL
FAIL	github.com/gohugoio/hugo/markup/goldmark/codeblocks	0.709s
FAIL
```
`.Type` became `Type: {foo=&#34;bar&#34;}`, so it is not empty.

I Will have the same result even if `attrStartIdx > 0` at L191 in `markup/goldmark/codeblocks/render.go`.

I suppose that the cause is https://github.com/gohugoio/hugo/pull/10119#discussion_r928178328

Reply:
>.Type became Type: {foo=&#34;bar&#34;}, so it is not empty.

It should be. 

This may be an existing issue, but it feels natural to fix as part of this PR. I suspect the simplest fix is to check for `{` where `Type` is evaluated. I think we can safely make that a reserved character not to be used in `Type`.

Reply:
Thanks for considering the direction.
I have checked for `{`.